### PR TITLE
Partially revert "mount: Check if src exists before mounted (#61752)"

### LINKED ISSUE #59183: mount module should fail when src doesn't exist when state is mounted
##### SUMMARY
If `src` doesn't exist in `mount` module when `state` is `mounted`, it still proceeds. It adds the entry to /etc/fstab like it would when `state` is `present`. But because it's `mounted` the task should fail.

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
ansible 2.8.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['~/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.3 (default, Jun 24 2019, 04:54:02) [GCC 9.1.0]
```

##### CONFIGURATION

```paste below
no output
```

##### OS / ENVIRONMENT

Debian 10

##### STEPS TO REPRODUCE

```yaml
- mount:
    src: /pathdoesnotexist
    path: /mnt
    opts: discard,nofail,defaults
    fstype: ext4
    state: mounted
```

##### EXPECTED RESULTS

It should give an error saying that `src` doesn't exist.

##### ACTUAL RESULTS

```paste below
ok: [5.6.7.8] => {
  "changed": false,
    "dump": "0",
    "fstab": "/etc/fstab",
    "fstype": "ext4",
    "invocation": {
        "module_args": {
            "backup": false,
            "boot": true,
            "dump": null,
            "fstab": null,
            "fstype": "ext4",
            "opts": "discard,nofail,defaults",
            "passno": null,
            "path": "/mnt",
            "src": "/pathdoesnotexist",
            "state": "mounted"
        }
    },
    "name": "/mnt",
    "opts": "discard,nofail,defaults",
    "passno": "0",
    "src": "/pathdoesnotexist"
}
```

### LINKED ISSUE #65855: Mount module fails on nfs mounts with "as it does not exist'
##### SUMMARY
When using the ansible 'mount' module to mount an NFS export, the play fails with an error containing "... as it does not exist".

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
Ansible module **mount**

##### ANSIBLE VERSION

```
ansible 2.10.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /var/data/src/ansible/lib/ansible
  executable location = /var/data/src/ansible/bin/ansible
  python version = 3.7.5 (default, Oct 17 2019, 12:16:48) [GCC 9.2.1 20190827 (Red Hat 9.2.1-1)]
```

##### CONFIGURATION

```
COLOR_VERBOSE(/etc/ansible/ansible.cfg) = dark gray
RETRY_FILES_ENABLED(/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Output of hostnamectl:
```
hostnamectl
   Static hostname: ansible-f31.example.com
         Icon name: computer-vm
           Chassis: vm
        Machine ID: 9af65fda237345db8f54e2fff640148b
           Boot ID: 88abdd71dcf048ea90680629998be20c
    Virtualization: vmware
  Operating System: Fedora 31 (Thirty One)
       CPE OS Name: cpe:/o:fedoraproject:fedora:31
            Kernel: Linux 5.3.15-300.fc31.x86_64
      Architecture: x86-64
```

##### STEPS TO REPRODUCE

It appears that commit d2f45dada7f46d5586f0175f5c1fce06a1b7dfe7 changed the **mount** module in such a way that will always fail when working with NFS remote mounts.

The commit resulted in code including the following snippet:
```
    elif state == 'mounted':
        if not os.path.exists(args['src']):
             module.fail_json(msg="Unable to mount %s as it does not exist" % args['src']) 
```
This would seem to always fail.  Simply commenting out lines **2** and **3** in the above snippet restores successful completion.

```yaml

```

##### EXPECTED RESULTS
When doing NFS mounts, Ansible should either use something like ```showmount` -e <nfsservername>``` to verify if the remote export exists, or skip validation entirely in the case of NFS mounts.

##### ACTUAL RESULTS
The play fails with an error including the text "as it does not exist".

```paste below

```

### LINKED ISSUE #67588: mount module cannot use `UUID=...` sources with `mounted` state
##### SUMMARY

According to the documentation the `mount` module can accept `UUID=...`-type strings as argument to its `src` parameter, see the third example in https://docs.ansible.com/ansible/latest/modules/mount_module.html. If the `UUID=...`-type argument is used together with a `state` of `present` as in the example, it works.
But when using a UUID-type argument with a `state` parameter of `mounted`, it results in a `Unable to mount %s as it does not exist` error. The corresponding code lines are: https://github.com/ansible/ansible/blob/3aae025ccef52bcca6d27e49f7755b3761e6662b/lib/ansible/modules/system/mount.py#L722-L723
This error check will never succeed for a UUID-type argument.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

mount
##### ANSIBLE VERSION

```paste below
devel branch at 052e8b7be4e4f8c6252b34b2a5b039c92af862f3
```

##### CONFIGURATION

```paste below
ANSIBLE_PIPELINING(/home/user/.ansible.cfg) = True
DEFAULT_FORKS(/home/user/.ansible.cfg) = 32
DEFAULT_HOST_LIST(/home/user/.ansible.cfg) = ['/home/user/ansible/inventory']
DEFAULT_ROLES_PATH(/home/user/.ansible.cfg) = ['/home/user/.ansible/roles']
```

##### OS / ENVIRONMENT

Host and Target both run Fedora 31 on x86_64

##### STEPS TO REPRODUCE

- modify the example below to use an existing device UUID on the target machine
- run the example task below and the error should show up

optional:
- change the argument of the `state` parameter from `mounted` to `present`
- run the example task and it should successfully add the entry to the `/etc/fstab` on the target system

```yaml
- name: mount /data Partition
  mount:
    path: /data
    src: "UUID=bca9c708-9c14-4fb9-ba69-1db6c7d597bb"
    fstype: "xfs"
    opts: "defaults,nofail"
    state: mounted
```

##### EXPECTED RESULTS

I expected the `mount` module to add the `UUID=...`-type entry to the `/etc/fstab` and mount the filesystem.

##### ACTUAL RESULTS

An error is thrown.

```paste below
fatal: [10.14.07.7]: FAILED! => {"changed": false, "msg": "Unable to mount UUID=bca9c708-9c14-4fb9-ba69-1db6c7d597bb as it does not exist"}

```

##### WORKAROUND
With Linux targets it's of course possible to use `/dev/disk/by-uuid/bca9c708-9c14-4fb9-ba69-1db6c7d597bb` as `src`-argument to make the error check succeed. But that's not exactly as the documentation makes you believe and it also does not create the canonical `UUID=...`-type entries in the `/etc/fstab`.

### LINKED ISSUE #67966: Mount module not mounting sshfs using state: mounted
##### SUMMARY
Ansible mount module not mounting sshfs using state: mounted

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
Ansible mount

##### ANSIBLE VERSION
```
ansible 2.9.5
  config file = /Users/ap/ansible-workspace/ansible-st-server/ansible.cfg
  configured module search path = ['/Users/ap/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/ap/anaconda3/envs/ansible/lib/python3.7/site-packages/ansible
  executable location = /Users/ap/anaconda3/envs/ansible/bin/ansible
  python version = 3.7.5 (default, Oct 25 2019, 10:52:18) [Clang 4.0.1 (tags/RELEASE_401/final)]
```

##### CONFIGURATION
```
DEFAULT_HOST_LIST(/Users/ap/ansible-workspace/ansible-st-server/ansible.cfg) = ['/Users/ap/ansible-workspace/ansible-st-server/hosts']
HOST_KEY_CHECKING(/Users/ap/ansible-workspace/ansible-st-server/ansible.cfg) = False
INTERPRETER_PYTHON(/Users/ap/ansible-workspace/ansible-st-server/ansible.cfg) = /usr/bin/python3
RETRY_FILES_ENABLED(/Users/ap/ansible-workspace/ansible-st-server/ansible.cfg) = False
```

##### OS / ENVIRONMENT
Target OS is Ubuntu 18.04.4 LTS

##### STEPS TO REPRODUCE
If you run the following code within a task, the fstab entry will be written correctly but it will not be mounted as expected. It works if root user mounts manually ( with ```mount /home/borg/backups-remote```) using a ssh connection to the server.

```
- name: Create dedicated user for borg backup
  user:
    name: borg
    comment: Dedicated user for borg backup
    password_lock: yes
    shell: /bin/bash
  register: borg_user
- name: Adding fstab entry for borg remote mount point
  mount:
    path: /home/borg/backups-remote
    src: borg@myserver.com:/users/borg
    fstype: fuse.sshfs
    opts: "defaults,allow_other,_netdev,idmap=user,follow_symlinks,identityfile=/home/borg/.ssh/id_rsa,allow_other,default_permissions,uid={{ borg_user.uid }},gid={{ borg_user.group }}"
    state: mounted
  register: backup_remote
```

##### EXPECTED RESULTS
Entry added to fstab (this works) and state mounted (this works not).

##### ACTUAL RESULTS
```
task path: /Users/ap/ansible-workspace/ansible-st-server/roles/docker_host_server/tasks/init_borg_backup_user.yml:48
<myserver.com> ESTABLISH SSH CONNECTION FOR USER: root
<myserver.com> SSH: EXEC sshpass -d59 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/Users/ap/.ansible/cp/e64dcd018c myserver.com '/bin/sh -c '"'"'echo ~root && sleep 0'"'"''
<myserver.com> (0, b'/root\n', b'OpenSSH_7.9p1, LibreSSL 2.7.3\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 48: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 17203\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<myserver.com> ESTABLISH SSH CONNECTION FOR USER: root
<myserver.com> SSH: EXEC sshpass -d59 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/Users/ap/.ansible/cp/e64dcd018c myserver.com '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456 `" && echo ansible-tmp-1583243356.797665-113017624172456="` echo /root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456 `" ) && sleep 0'"'"''
<myserver.com> (0, b'ansible-tmp-1583243356.797665-113017624172456=/root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456\n', b'OpenSSH_7.9p1, LibreSSL 2.7.3\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 48: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 17203\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
Using module file /Users/ap/anaconda3/envs/ansible/lib/python3.7/site-packages/ansible/modules/system/mount.py
<myserver.com> PUT /Users/ap/.ansible/tmp/ansible-local-17190njygosda/tmpllr8128c TO /root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456/AnsiballZ_mount.py
<myserver.com> SSH: EXEC sshpass -d59 sftp -o BatchMode=no -b - -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/Users/ap/.ansible/cp/e64dcd018c '[myserver.com]'
<myserver.com> (0, b'sftp> put /Users/ap/.ansible/tmp/ansible-local-17190njygosda/tmpllr8128c /root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456/AnsiballZ_mount.py\n', b'OpenSSH_7.9p1, LibreSSL 2.7.3\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 48: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 17203\r\ndebug3: mux_client_request_session: session request sent\r\ndebug2: Remote version: 3\r\ndebug2: Server supports extension "posix-rename@openssh.com" revision 1\r\ndebug2: Server supports extension "statvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "fstatvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "hardlink@openssh.com" revision 1\r\ndebug2: Server supports extension "fsync@openssh.com" revision 1\r\ndebug3: Sent message fd 3 T:16 I:1\r\ndebug3: SSH_FXP_REALPATH . -> /root size 0\r\ndebug3: Looking up /Users/ap/.ansible/tmp/ansible-local-17190njygosda/tmpllr8128c\r\ndebug3: Sent message fd 3 T:17 I:2\r\ndebug3: Received stat reply T:101 I:2\r\ndebug1: Couldn\'t stat remote file: No such file or directory\r\ndebug3: Sent message SSH2_FXP_OPEN I:3 P:/root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456/AnsiballZ_mount.py\r\ndebug3: Sent message SSH2_FXP_WRITE I:4 O:0 S:32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 4 32768 bytes at 0\r\ndebug3: Sent message SSH2_FXP_WRITE I:5 O:32768 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:6 O:65536 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:7 O:98304 S:15223\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 5 32768 bytes at 32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 6 32768 bytes at 65536\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 7 15223 bytes at 98304\r\ndebug3: Sent message SSH2_FXP_CLOSE I:4\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<myserver.com> ESTABLISH SSH CONNECTION FOR USER: root
<myserver.com> SSH: EXEC sshpass -d59 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/Users/ap/.ansible/cp/e64dcd018c myserver.com '/bin/sh -c '"'"'chmod u+x /root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456/ /root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456/AnsiballZ_mount.py && sleep 0'"'"''
<myserver.com> (0, b'', b'OpenSSH_7.9p1, LibreSSL 2.7.3\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 48: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 17203\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<myserver.com> ESTABLISH SSH CONNECTION FOR USER: root
<myserver.com> SSH: EXEC sshpass -d59 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/Users/ap/.ansible/cp/e64dcd018c -tt myserver.com '/bin/sh -c '"'"'/usr/bin/python3 /root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456/AnsiballZ_mount.py && sleep 0'"'"''
<myserver.com> (0, b'\r\n{"changed": true, "name": "/home/borg/backups-remote", "opts": "defaults,allow_other,_netdev,idmap=user,follow_symlinks,identityfile=/home/borg/.ssh/id_rsa,allow_other,default_permissions,uid=1001,gid=1001", "dump": "0", "passno": "0", "fstab": "/etc/fstab", "src": "borg@myserver.com:/users/borg", "fstype": "fuse.sshfs", "invocation": {"module_args": {"path": "/home/borg/backups-remote", "src": "borg@myserver.com:/users/borg", "fstype": "fuse.sshfs", "opts": "defaults,allow_other,_netdev,idmap=user,follow_symlinks,identityfile=/home/borg/.ssh/id_rsa,allow_other,default_permissions,uid=1001,gid=1001", "state": "mounted", "boot": true, "backup": false, "dump": null, "fstab": null, "passno": null}}}\r\n', b'OpenSSH_7.9p1, LibreSSL 2.7.3\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 48: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 17203\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to myserver.com closed.\r\n')
<myserver.com> ESTABLISH SSH CONNECTION FOR USER: root
<myserver.com> SSH: EXEC sshpass -d59 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'User="root"' -o ConnectTimeout=10 -o ControlPath=/Users/ap/.ansible/cp/e64dcd018c myserver.com '/bin/sh -c '"'"'rm -f -r /root/.ansible/tmp/ansible-tmp-1583243356.797665-113017624172456/ > /dev/null 2>&1 && sleep 0'"'"''
<myserver.com> (0, b'', b'OpenSSH_7.9p1, LibreSSL 2.7.3\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 48: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 17203\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
changed: [myserver.com] => {
    "changed": true,
    "dump": "0",
    "fstab": "/etc/fstab",
    "fstype": "fuse.sshfs",
    "invocation": {
        "module_args": {
            "backup": false,
            "boot": true,
            "dump": null,
            "fstab": null,
            "fstype": "fuse.sshfs",
            "opts": "defaults,allow_other,_netdev,idmap=user,follow_symlinks,identityfile=/home/borg/.ssh/id_rsa,allow_other,default_permissions,uid=1001,gid=1001",
            "passno": null,
            "path": "/home/borg/backups-remote",
            "src": "borg@myserver.com:/users/borg",
            "state": "mounted"
        }
    },
    "name": "/home/borg/backups-remote",
    "opts": "defaults,allow_other,_netdev,idmap=user,follow_symlinks,identityfile=/home/borg/.ssh/id_rsa,allow_other,default_permissions,uid=1001,gid=1001",
    "passno": "0",
    "src": "borg@myserver.com:/users/borg"
}
```

### PR DESCRIPTION
**Pull request by patrakov**
This reverts part of commit 72023d7462e78635264fd12bfdb23894b4163cba.

The immediate reason is that it breaks mounts where src is not a path.
Examples of such mounts are nfs, cifs, glusterfs, ceph, tmpfs,
overlayfs, and it is too hard to come with an exhaustive list,
especially if we take non-Linux systems into account.

Additionally, it did not really fix the issue (#59183) that it intended
to fix, because the mount could fail but leave a non-working fstab entry
for reasons other than non-existing src path.

##### SUMMARY

This was originally submitted as #65869, with some extra commits that attempt to re-fix #59183. However, that pull request was not properly reviewed in 2+ months, presumably because the additional commits are too complex. Therefore, I am resubmitting just the minimal fix that is hopefully easier to review.

Fixes: #65855
Fixes: #67588
Fixes: #67966

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
mount

##### ADDITIONAL INFORMATION
See #65544 and #65869.

### CODE REVIEW

**Comment by bcoca** [comment]
I would prefer the fix would add an aditional check `not startswith('UUID=', 'LABEL=') ` and then keep the path.exists check for those cases

**Comment by patrakov** [comment]
I am absolutely opposed to any additional checks. We have already tried that while discussing #65544 and failed. Martin initially missed CIFS. I missed the UUID and LABEL cases. Then OpenBSD has special syntax (which looks like `55128c3700af5491.a`) too, which is described in section "DUIDs and /etc/fstab" in the Absolute OpenBSD book. I am sure we miss many other cases.

Finally, missing src is not the only reason why a mount could fail. E.g., there could have been a wrong filesystem, or the kernel has not been compiled with support for the correct filesystem, or there are incorrect options, or ... Therefore, checking src does not, by itself, fix the bug that it was meant to fix. Even if we do check src, we would need to write code that deals with any unexplained mount failures. And once we have this code (see #65969), checking src becomes unnecessary.

In summary, we should not go into the business of predicting whether the mount command will succeed. Just let it fail when it needs to fail and then deal with the consequences.

**Comment by patrakov** [comment]
Also, why is it P3? It is a fix for a critical regression (cannot mount remote filesystems at all) in the mount module. One of the rationales for this merge request is exactly to revert the regression as soon as possible, because it is far worse than the bug that the commit 72023d7 tried to fix.

**Comment by patrakov** [comment]
> I would prefer the fix would add an aditional check `not startswith('UUID=', 'LABEL=') ` and then keep the path.exists check for those cases

Done, see #68155, but I would very much prefer this pull request (and ideally #65869) to be merged instead of that.

**Comment by Akasurde** [comment]
First of all, sorry for introducing the wrong fix. I agree with @patrakov, we should not predict anything about the mount and just remove this check once.

**Comment by patrakov** [comment]
@Akasurde could you please review the proper fix (i.e. the attempt to remove created directories and restore the fstab if the mount fails) in #65869 ?

**Comment by Akasurde** [comment]
> @Akasurde could you please review the proper fix (i.e. the attempt to remove created directories and restore the fstab if the mount fails) in #65869 ?

Code LGTM, I will test and comment on #65869. I would see this PR getting merged first.

**Comment by Akasurde** [comment]
superseded by https://github.com/ansible-collections/ansible.posix/pull/33

**Comment by Akasurde** [comment]
@patrakov Thanks for the contribution.
